pkg/gardenlet/controller/seed/seed/components.go

```diff
@@ -242,10 +243,10 @@ func defaultDependencyWatchdogs(
 				kubeapiserver.NewDependencyWatchdogProberConfiguration,
 			}
 			dependencyWatchdogProberConfiguration = proberapi.Config{
-				InternalKubeConfigSecretName: dependencywatchdog.InternalProbeSecretName,
-				ExternalKubeConfigSecretName: dependencywatchdog.ExternalProbeSecretName,
-				ProbeInterval:                &metav1.Duration{Duration: dependencywatchdog.DefaultProbeInterval},
-				DependentResourceInfos:       make([]proberapi.DependentResourceInfo, 0, len(dependencyWatchdogProberConfigurationFuncs)),
+				KubeConfigSecretName:        dependencywatchdog.KubeConfigSecretName,
+				ProbeInterval:               &metav1.Duration{Duration: dependencywatchdog.DefaultProbeInterval},
+				DependentResourceInfos:      make([]proberapi.DependentResourceInfo, 0, len(dependencyWatchdogProberConfigurationFuncs)),
+				KCMNodeMonitorGraceDuration: metav1.Duration{Duration: kubecontrollermanager.NodeMonitorGraceDurationK8sGreaterEqual127},
```

Comment:
What if the cluster is using Kubenretes < 1.27? Then a wrong value will be passed here? What if https://github.com/gardener/gardener/blob/bb4e628270195aae6d698f7bf325268d9d5e8553/example/90-shoot.yaml#L218 is configured by the user? Again, a wrong value will be passed here?

Reply:
If it is configured by the user, then DWD will pick it up from the shoot. See [this](https://github.com/gardener/dependency-watchdog/blob/799d930ef62c6b8e85c9ee579c169a10b6f67d43/controllers/cluster/cluster_controller.go#L192-L199).

Reply:
OK. What about this case?

> What if the cluster is using Kubernetes < 1.27?

Reply:
>What if the cluster is using Kubernetes < 1.27?

IMO, for this case also we should keep the value to `40s` as it is the k8s default, but if g/g uses `2m` we can add a check for this.

Reply:
Hm, why would we keep the default value even if the cluster is configured differently? Can you please help me understand this?

Reply:
There is an admission plugin that always sets the `NodeMonitoringGracePeriod` to shoots (ref: https://github.com/gardener/gardener/blob/master/plugin/pkg/shoot/validator/admission.go#L795-L801) hence dwd will always pick up from shoots

Reply:
Then we remove this `KCMNodeMonitorGraceDuration` entirely to prevent confusion/discussions like this?

Reply:
Agreed. That makes sense

Reply:
`NodeMonitoringGracePeriod` is a pointer to a duration in the shoot spec. If there is a change done in the admission plugin causing it not to set the value of `NodeMonitorGracePeriod`, then DWD probes will fail to start as they have a hard dependency on the `KCMNodeMonitorGraceDuration`. So we need to make sure that the admission plugin never has an incompatible change where the `NodeMonitorGracePeriod` is not set. If this can be ensured then we can make this change.

Reply:
Ok. I will make a patch in DWD with the following change :-
Make the `KCMNodeMonitorGraceDuration` an optional field in the config. Default the value for `KCMNodeMonitorGraceDuration`, if not present in the shoot, as per its k8s version.

Reply:
This is done and available with `dwd:v1.2.1`